Mathlib/Algebra/Algebra/Subalgebra/Unitization.lean

```diff
@@ -258,33 +306,90 @@ theorem StarSubalgebra.toNonUnitalStarSubalgebra_toStarSubalgebra (S : StarSubal
+variable [Algebra R A] [StarModule R A] -/
+
 /-- The natural star `R`-algebra homomorphism from the unitization of a non-unital star subalgebra
 to its `StarSubalgebra.adjoin`. -/
-def NonUnitalStarSubalgebra.unitization (S : NonUnitalStarSubalgebra R A) :
-    Unitization R S →⋆ₐ[R] StarSubalgebra.adjoin R (S : Set A) :=
-  Unitization.starLift
-    { toFun := Subtype.map (p := (· ∈ S)) (q := (· ∈ StarSubalgebra.adjoin R (S : Set A))) id <|
-        fun _x hx => StarSubalgebra.subset_adjoin R _ hx
-      map_smul' := fun (_ : R) _ => rfl
-      map_zero' := rfl
-      map_add' := fun _ _ => rfl
-      map_mul' := fun _ _ => rfl
-      map_star' := fun _ => rfl }
+def unitization : Unitization R s →⋆ₐ[R] A :=
+  Unitization.starLift <| NonUnitalStarSubalgebraClass.subtype s
 
 @[simp]
-theorem NonUnitalStarSubalgebra.unitization_apply_coe (S : NonUnitalStarSubalgebra R A)
-    (x : Unitization R S) :
-    (S.unitization x : A) = algebraMap R (StarSubalgebra.adjoin R (S : Set A)) x.fst + x.snd :=
+theorem unitization_apply (x : Unitization R s) : unitization s x = algebraMap R A x.fst + x.snd :=
   rfl
 
-theorem NonUnitalStarSubalgebra.unitization_surjective (S : NonUnitalStarSubalgebra R A) :
-    Function.Surjective S.unitization :=
-  have : StarSubalgebra.adjoin R S ≤
-      ((StarSubalgebra.adjoin R (S : Set A)).subtype.comp S.unitization).range :=
-    StarSubalgebra.adjoin_le fun a ha ↦ ⟨(⟨a, ha⟩ : S), by simp⟩
-  fun x ↦ match this x.property with | ⟨y, hy⟩ => ⟨y, Subtype.ext hy⟩
+theorem unitization_range : (unitization s).range = StarSubalgebra.adjoin R s := by
+  rw [unitization, Unitization.starLift_range]
+  simp only [NonUnitalStarAlgHom.coe_range, NonUnitalStarSubalgebraClass.coeSubtype,
+    Subtype.range_coe_subtype]
+  rfl
 
-end StarSubalgebra
+end Semiring
+
+section Field
+
+variable {R S A : Type*} [Field R] [StarRing R] [Ring A] [StarRing A] [Algebra R A]
+  [StarModule R A] [SetLike S A] [hSA : NonUnitalSubringClass S A] [hSRA : SMulMemClass S R A]
+  [StarMemClass S A] (s : S)
+
+theorem unitization_injective (h1 : (1 : A) ∉ s) : Function.Injective (unitization s) :=
+  AlgHomClass.unitization_injective s h1 (unitization s) fun _ ↦ by simp
+
+/-- If a `NonUnitalStarSubalgebra` over a field does not contain `1`, then its unitization is
+isomorphic to its `StarSubalgebra.adjoin`. -/
+@[simps! apply_coe]
+noncomputable def unitizationStarAlgEquiv (h1 : (1 : A) ∉ s) :
+    Unitization R s ≃⋆ₐ[R] StarSubalgebra.adjoin R (s : Set A) :=
+  let starAlgHom : Unitization R s →⋆ₐ[R] StarSubalgebra.adjoin R (s : Set A) :=
+    ((unitization s).codRestrict _
+      fun x ↦ (unitization_range s).le <| Set.mem_range_self x)
+  StarAlgEquiv.ofBijective starAlgHom <| by
+    refine ⟨?_, fun x ↦ ?_⟩
+    · have h_eq : unitization s = (StarSubalgebra.subtype _).comp starAlgHom := rfl
+      have this := unitization_injective s h1
+      rw [h_eq, StarAlgHom.coe_comp] at this
+      exact this.of_comp
```

Comment:
Quick note: I was surprised that the proof was different than in the non-star case, which lead me to discover a small hole in the API (see #6668). But there's really no point in enforcing similar proofs for four quick lines so of course you don't have to wait for that.

Reply:
Thanks for plugging that hole for me!